src/components/runbooks/RunbookPage.js

```diff
@@ -0,0 +1,260 @@
+import {Link, withRouter} from 'react-router';
+import {observable, computed, action, transaction} from 'mobx';
+import {observer, inject} from 'mobx-react';
+import {isEqual, pick, forEach, some} from 'lodash';
+
+import {RUNBOOK_RUN_STATUSES} from '../../consts';
+import {Runbook} from '../../stores/Runbooks';
+import {RunbookRun} from '../../stores/RunbookRuns';
+import RunbookForm from './RunbookForm';
+
+@withRouter
+@inject('runbooks', 'runbookRuns')
+@observer
+export default class RunbookPage extends Component {
+  static async fetchData(
+    {runbooks, runbookRuns, params: {runbookId}}
+  ) {
+    //const runbookUrl = `/api/v1/runbooks/${runbookId}`;
+    //const runbookResponse = await fetch(runbookUrl);
+    //const runbookResponseBody = await runbookResponse.json();
+    const runbookResponseBody = {
+      id: runbookId,
+      description: 'Demo runbook description',
+      name: 'Demo runbook',
+      runbook: 'IyEvYmluL2Jhc2gKCmVjaG8gIkhlbGxvIFdvcmxkISIK',
+      type: 'bash',
+      tags: ['Monitoring'],
+      status: 'scheduled',
+      regionId: 'east-3.hooli.net'
+    };
+    runbooks.items = [new Runbook(runbookResponseBody)];
+
+    //const runbookRunsUrl = `/api/v1/runbook_runs?runbook=${runbookId}`;
+    //const runbookRunsResponse = await fetch(runbookRunsUrl);
+    //const runbookRunsResponseBody = await runbookRunsResponse.json();
+    const runbookRunsResponseBody = {
+      runs: [
+        {
+          id: 434,
+          updated_at: null,
+          created_at: '2016-12-20T16:18:42.150142',
+          status: 'scheduled',
+          runbook: {
+            id: '602',
+            name: 'Demo runbook',
+            tags: ['Monitoring']
+          },
+          regionId: 'region_one',
+          user: 'cloud_user'
+        },
+        {
+          id: 251,
+          updated_at: '2016-12-20T16:33:42.150344',
+          created_at: '2016-12-20T16:18:42.150312',
+          status: 'failed',
+          runbook: {
+            id: '602',
+            name: 'Demo runbook',
+            tags: ['Monitoring']
+          },
+          regionId: 'region_one',
+          user: 'cloud_user'
+        },
+        {
+          id: 248,
+          updated_at: '2016-12-20T16:33:42.150344',
+          created_at: '2016-12-20T16:18:42.150385',
+          status: 'finished',
+          runbook: {
+            id: '602',
+            name: 'Demo runbook',
+            tags: ['Monitoring']
+          },
+          regionId: 'region_one',
+          user: 'cloud_user'
+        }
+      ]
+    };
+    runbookRuns.items = runbookRunsResponseBody.runs.map((run) => new RunbookRun(run));
+  }
+
+  @observable key = Date.now();
+  @observable data;
+  @observable errors = {};
+
+  @action
+  componentWillMount() {
+    const {params: {runbookId}, runbooks} = this.props;
+    this.data = new Runbook(runbooks.get(runbookId));
+  }
+
+  @action
+  onChange = (e) => {
+    const {name, value} = e.target;
+    transaction(() => {
```

Comment:
It's not necessary to use transaction withing action - all actions are transactions